.github/workflows/pr-verify.yml

```diff
@@ -161,3 +161,31 @@ jobs:
 
       - name: Build Rust Docs
         run: RUSTDOCFLAGS="--enable-index-page --check -Zunstable-options" cargo doc --no-deps --all-features
+  
+  build_gradlew_jvm:
+    name: Build and Publish JVM
+    runs-on: ubuntu-latest
+    needs: [verify, test, lint, build]
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+      - name: Build JNI package
+        run: make build-jni
+      - name: Build and Publish with Gradle
+        working-directory: java
+        run: ./gradlew build
+      - name: Publish to GitHub Packages
+        working-directory: java
+        run: ./gradlew publish
```

Comment:
should we keep the build and publish under the same job or separate it to allow more manual publish?

Reply:
Ideally we would update the version of snapshot before releasing new one so that might take care of it for now